src/docfx/build/dependency/Resolve.cs

```diff
@@ -46,14 +46,18 @@ public static string GetLink(string path, object relativeTo, object resultRelati
             return link;
         }
 
-        public static XrefSpec ResolveXref(string uid, XrefMap xrefMap, Document file, DependencyMapBuilder dependencyMapBuilder, string moniker = null)
+        public static (XrefSpec spec, Document referencedFile) ResolveXref(string uid, XrefMap xrefMap, Document file, DependencyMapBuilder dependencyMapBuilder, string moniker = null)
```

Comment:
Can this return the already transformed `XrefSpec` so caller don't have to base the link themselves?

Reply:
One option is that we create a new instance of `XrefSpec` since we need to output the original one later. Another option is the consumer rebase the link themselves, we want to minimize the memory, so prefer to just return it here.

Reply:
The contract does not hold if [Xref] resolves to an `XrefSpec` instead of `href` in the future.